crates/router/src/core/revenue_recovery/types.rs

```diff
@@ -563,18 +645,94 @@ impl Action {
         match response {
             Ok(_payment_data) => match payment_attempt.status.foreign_into() {
                 RevenueRecoveryPaymentsAttemptStatus::Succeeded => {
+                    let payment_intent = db
+                        .find_payment_intent_by_id(
+                            &(state).into(),
+                            &payment_attempt.payment_id,
+                            &revenue_recovery_payment_data.key_store,
+                            revenue_recovery_payment_data
+                                .merchant_account
+                                .storage_scheme,
+                        )
+                        .await
+                        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)
+                        .change_context(errors::RecoveryError::PaymentCallFailed)
+                        .attach_printable(
+                            "Failed to fetch payment intent for calculate workflow reopening",
+                        )?;
+
+                    let connector_customer_id = payment_intent
+                        .extract_connector_customer_id_from_payment_intent()
+                        .change_context(errors::RecoveryError::ValueNotFound)
+                        .attach_printable("Failed to extract customer ID from payment intent")?;
+
+                    let payment_processor_token = process
+                        .tracking_data
+                        .clone()
+                        .parse_value::<storage::revenue_recovery::RevenueRecoveryWorkflowTrackingData>("RevenueRecoveryWorkflowTrackingData")
+                        .change_context(errors::RecoveryError::ValueNotFound)?
+                        .active_token
+                        .ok_or(errors::RecoveryError::ValueNotFound)
+                        .change_context(errors::RecoveryError::ValueNotFound)?
+                        .payment_processor_token;
+
+                    storage::revenue_recovery_redis_operation::RedisTokenManager::update_payment_processor_token_error_code_retry_count(
+                        state,
+                        &connector_customer_id,
+                        &payment_processor_token,
+                        "-1".to_string()
+                    );
+
+                    // unlocking the token
+                    storage::revenue_recovery_redis_operation::RedisTokenManager::unlock_connector_customer_status(
+                        state,
+                        &connector_customer_id, 
+                    )
+                    .await;
+
+                    update_calculate_workflow_on_success(state, &payment_intent).await?;
+
                     Ok(Self::SuccessfulPayment(payment_attempt))
                 }
                 RevenueRecoveryPaymentsAttemptStatus::Failed => {
-                    Self::decide_retry_failure_action(
+                    let payment_intent = db
+                        .find_payment_intent_by_id(
```

Comment:
We can't read payment related database calls, which against the modularity, refer with @Aprabhat19